### PR TITLE
Various refactoring and potential bugfixes

### PR DESCRIPTION
**Pull request by SteffenGorenflo**

### CODE REVIEW

File: src/plugin.ts [hunk 1]
```diff
@@ -213,6 +217,30 @@ class User {
...
+                tags: ['api', 'user'],
+                validate: {
+                    params: {
+                        userid: this.joi.string()
+                            .required()
+                            .description('User Id')
+                    },
+                    payload: this.joi.object().keys({
+                        // TODO: verify mail pattern?
+                        mail: this.joi.string().required()
```
Comment (timoweiss) [inline src/plugin.ts]: https://github.com/hapijs/joi#stringemailoptions

File: src/plugin.ts [hunk 2]
```diff
@@ -352,20 +381,24 @@ class User {
...
 
+                    var newUser = {
+                        password: hash,
+                        strategy: 'default',
+                        uuid: this.uuid.v4(),
+                        verified: false,
+
+                        // TODO: retrieve picture from own database
+                        // dummy picture
+                        picture: {
```
Comment (timoweiss) [inline src/plugin.ts]: The frontend should ship the dummy picture. It should be displayed if property `picture` is not present.

File: src/plugin.ts [hunk 3]
```diff
@@ -428,11 +461,40 @@ class User {
...
+        });
+    };
+
+    /**
      * Delete user by user id.
      * @param request
      * @param reply
      */
     private deleteUser = (request, reply) => {
+        if (request.params.userid != request.auth.credentials._id) {
```
Comment (timoweiss) [inline src/plugin.ts]: !==

File: src/plugin.ts (hunk 2 above)
Comment (SteffenGorenflo) [inline src/plugin.ts]: we had that discussion before and we decided not to

File: src/plugin.ts (hunk 1 above)
Comment (SteffenGorenflo) [inline src/plugin.ts]: verify in backend or frontend?

File: src/plugin.ts (hunk 2 above)
Comment (timoweiss) [inline src/plugin.ts]: and how are you gonna solve this?

File: src/plugin.ts (hunk 1 above)
Comment (timoweiss) [inline src/plugin.ts]: both

Comment (SteffenGorenflo) [inline src/plugin.ts]: do it, feel free :)

File: src/plugin.ts (hunk 2 above)
Comment (SteffenGorenflo) [inline src/plugin.ts]: aight, picture field will be undefined per default

**Pull request merged**